component/src/main/java/com/celements/photo/container/ImageUrlDim.java

```diff
@@ -0,0 +1,42 @@
+package com.celements.photo.container;
+
+import com.google.common.base.Optional;
+
+import groovy.lang.Immutable;
+
+@Immutable
+public class ImageUrlDim {
```

Comment:
I'm not sure I understood the intention, but factory usually is used to handle polymorphism and builder to fight long or heavily overloaded constructors. We have neither here.